### PR TITLE
Minor fixes to resolve bugs introduced in TMOP

### PR DESCRIPTION
**Pull request by kmittal2**
Resolve bugs that have come up due to #1922.  

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#2462](https://github.com/mfem/mfem/pull/2462) | @kmittal2 | @tzanio | @tzanio + @v-dobrev + @camierjs | 08/09/21 | 08/10/21 | ⌛due 08/17/21 | |

### CODE REVIEW

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Review by tzanio — APPROVED** [review]

**Comment by tzanio** [comment]
@camierjs and @v-dobrev -- this is a hotfix for master, so a fast review will be appreciated

**Review by v-dobrev — APPROVED** [review]

**Comment by v-dobrev** [comment]
LGTM

File: fem/tmop.cpp [hunk 1]
```diff
@@ -1336,6 +1336,7 @@ void DiscreteAdaptTC::FinalizeParDiscreteTargetSpec(const ParGridFunction &t)
                                            ptspec_fes->FEColl(), ncomp);
 
    delete tspec_pgf;
+   tspec.HostReadWrite();
```
Comment (camierjs) [inline fem/tmop.cpp:1339]: Would it be better to address the issue inside `ParGridFunction(ParFiniteElementSpace *pf, GridFunction *gf);`,
where `gf->GetData()` might be the cause?

Comment (camierjs) [reply fem/tmop.cpp:1339]: The error can also be caused by the `double*` cast to use `GridFunction(FiniteElementSpace *f, double *data)`.

Comment (v-dobrev) [reply fem/tmop.cpp:1339]: > The error can also be caused by the `double*` cast to use `GridFunction(FiniteElementSpace *f, double *data)`.

I think this is the issue (the second argument is a `Vector` implicitly cast to `double *`), so it cannot be solved inside the `(Par)GridFunction` constructor.

**Comment by v-dobrev** [comment]
LGTM

**Comment by v-dobrev** [comment]
@camierjs, take a look at the updates I just pushed -- this should be closer to what we discussed earlier today.

**Review by tzanio — APPROVED** [review]
Thank you @v-dobrev

**Review by camierjs — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**